File: frontend/src/lib/tauri-api/review-summary.ts
import type { WorkspaceReviewSummary } from '../../types/review-summary';
import { invokeCommand } from './client';

export function getWorkspaceReviewSummary(workspaceId: string): Promise<WorkspaceReviewSummary> {
  return invokeCommand<WorkspaceReviewSummary>('get_workspace_review_summary', { workspaceId });
}

export function refreshWorkspaceReviewSummary(workspaceId: string): Promise<WorkspaceReviewSummary> {
  return invokeCommand<WorkspaceReviewSummary>('refresh_workspace_review_summary', { workspaceId });
}

File: src-tauri/src/commands/review_summary.rs
use tauri::State;

use crate::models::WorkspaceReviewSummary;
use crate::services::review_summary_service;
use crate::state::AppState;

#[tauri::command]
pub fn get_workspace_review_summary(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceReviewSummary, String> {
    review_summary_service::get_workspace_review_summary(&state, &workspace_id)
}

#[tauri::command]
pub fn refresh_workspace_review_summary(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceReviewSummary, String> {
    review_summary_service::refresh_workspace_review_summary(&state, &workspace_id)
}
